packages/react-components/react-virtualizer/src/hooks/useIntersectionObserver.ts

```diff
@@ -1,8 +1,40 @@
 import type { Dispatch, MutableRefObject, SetStateAction } from 'react';
 import * as React from 'react';
 import { useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
 
-const { useState, useRef } = React;
+const { useCallback, useState, useRef } = React;
+import { useMutationObserver } from './useMutationObserver';
+
+/**
+ * This function will take the rootMargin and flip the sides if we are in RTL based on the computed reading direction of the target element.
+ * @param ltrRootMargin the margin to be processed and flipped if required
+ * @param target target element that will have its current reading direction determined
+ * @returns the corrected rootMargin (if it was necessary to correct)
+ */
+export const getRTLRootMargin = (ltrRootMargin: string, target?: Element | Document | null | undefined): string => {
+  if (target) {
+    // get the computed dir for the target element
+    const newDir = getComputedStyle(target as Element).direction;
```

Comment:
Maybe I am blind, but I can't see where this function is defined 😅

Reply:
Were you supposed to get it from the Window object or something?

Reply:
Well it must be working! 😄

Reply:
Is it something that will be available in SSR as well?

Reply:
It's a `window` function so ya you're right for SSR we need to check if this is there. If not we shouldn't call it and bail.

Reply:
We'll have the same issues with IO and MO here. Granted, a virtualizer should basically do nothing in SSR in the first place other than naively render the first bit. @Mitch-At-Work there's probably something there to check out and validate (not sure it fits this PR though)